src/Database/LSMTree/Run/Index/Compact.hs

```diff
@@ -0,0 +1 @@
+module Database.LSMTree.Run.Index.Compact () where
```

Comment:
We only plan to have one index, so I don't think we need extra hierarchy here.
```suggestion
module Database.LSMTree.Run.IndexCompact () where
```
or can just call it `Database.LSMTree.Run.Index`

Reply:
In the future, might it not make sense to have a full index variant too? Not for the Cardano use case, but for the more general use case

Reply:
True, but even there we probably don't need a lot of internal hierarchy.